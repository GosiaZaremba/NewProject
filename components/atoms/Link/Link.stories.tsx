import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { Link } from './Link';
import { text } from '@storybook/addon-knobs';

storiesOf('atoms/Link', module)
    .addDecorator((story) => story())
    .add('default', () => (
        // eslint-disable-next-line @typescript-eslint/no-empty-function
        <Link text={text('Link text', 'Link text')} onPress={() => {}} />
    ));
