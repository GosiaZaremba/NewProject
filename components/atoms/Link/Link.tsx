import React from 'react';
import {Text} from 'react-native';
import styles from './Link.styles';

export type Props = {
  text: string;
  onPress: () => void;
};

export const Link: React.FC<Props> = ({text, onPress}) => {
  return (
    <Text onPress={onPress} style={styles.text}>
      {text}
    </Text>
  );
};
