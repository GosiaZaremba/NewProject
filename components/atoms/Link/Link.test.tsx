import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';
import '@testing-library/jest-native';
import renderer from 'react-test-renderer';
import { Link } from './Link';

const mockOnPress = jest.fn();

describe('Atom Link', () => {
    test('Component Snapshot', () => {
        const tree = renderer
            .create(
                <Link
                    text={'Link text'}
                    testID={'atom-link'}
                    onPress={mockOnPress}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('ensure variants works correctly', () => {
        const { getByTestId } = render(
            <Link
                text={'Link text'}
                testID={'atom-link'}
                onPress={mockOnPress}
            />
        );

        const link = getByTestId('atom-link');
        expect(link).toHaveTextContent('Link text');
        fireEvent.press(link);
        expect(mockOnPress).toHaveBeenCalledTimes(1);
    });
});
