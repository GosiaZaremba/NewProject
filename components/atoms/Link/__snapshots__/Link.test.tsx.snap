// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom Link Component Snapshot 1`] = `
<Text
  onPress={[MockFunction]}
  style={
    Object {
      "color": "black",
      "textDecorationLine": "underline",
    }
  }
  testID="atom-link"
>
  Link text
</Text>
`;
